test/Datadog.Trace.Tests/Sampling/RuleBasedSamplerTests.cs

```diff
@@ -88,6 +89,37 @@ public void No_Registered_Rules_Uses_Legacy_Rates()
                 0.05f);
         }
 
+        [Fact]
+        public void Zero_rate_filters_everything()
+        {
+            var oldGenerator = SpanIdGenerator.ThreadInstance;
+
+            // Create an id generator that will always return 0
+            // This is needed because when creating a SpanContext with an id of 0,
+            // the value is considered as "no value" and the id generator is called
```

Comment:
Any reason why you didn't create the `SpanContext` with 1 instead of 0?

(side note: this workaround makes me think we should add an optional `IIdGenerator` into the `SpanContext` ctor for texting purposes and default to the thread-static instance)

Reply:
Never mind. The scenario you were testing was specifically id == 0 🤦🏽